modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/permission/PermissionUtil.java

```diff
@@ -81,6 +84,57 @@ public static List<ResourcePermission> getResourcePermissions(
 			String.valueOf(resourceId));
 	}
 
+	public static List<Permission> getResourcePermissionsByResourceName(
+			long companyId, List<ResourceAction> resourceActions,
+			long resourceId, String resourceName,
+			ResourcePermissionLocalService resourcePermissionLocalService,
+			RoleLocalService roleLocalService, String[] roleNames)
+		throws PortalException {
+
+		_checkResources(
+			companyId, resourceId, resourceName,
+			resourcePermissionLocalService);
+
+		List<ResourcePermission> resourcePermissions =
+			resourcePermissionLocalService.getResourcePermissions(resourceName);
+
+		List<Permission> permissions = new ArrayList<>();
+
+		Map<String, Permission> permissionMap = new HashMap<>();
+
+		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Role role = roleLocalService.getRole(
+				resourcePermission.getRoleId());
+
+			if ((roleNames == null) ||
+				((roleNames != null) &&
```

Comment:
I think that the `roleNames != null` check is not necessary as we first check if `roleName == null`

Reply:
> I was reviewing this PR and there is something that could be a problem 😢
> 
> I made some tests and this same to fix the problem for the entries and the lists, I mean, it returns the proper information for both endpoints
> 
>     * List `http://localhost:8080/o/headless-delivery/v1.0/sites/{siteId}/blog-postings/permissions'`
> 
>     * Entries [http://localhost:8080/o/headless-delivery/v1.0/blog-postings/{blogPostingId}permissions](http://localhost:8080/o/headless-delivery/v1.0/blog-postings/%7BblogPostingId%7Dpermissions)
> 
> 
> The proper solution (I know this is a draft, I didn't expect the proper solution here) should be to add this base implementation to the rest builder and generate the code for all APIs. So, where is the problem?
> 
> The problem is in the new method added to vulcan-api. Again, we are breaking backward compatibility.
> 
> I was checking if we could add this logic to an existing method so we didn't break compatibility and, if a client notices that the endpoints contain a bug, should install an FP/SP/Hotfix, like any other bug.

We discussed the same. Can't we add the method and then backport?